File: USACO/2021/Feburary/Bronze/yoc.py
zod = "Ox, Tiger, Rabbit, Dragon, Snake, Horse, Goat, Monkey, Rooster, Dog, Pig, Rat".split(", ")
d = {"Bessie": 1200}
n = int(input())
for i in range(n):
    cmd = input().split(" ")
    new = cmd[0]
    yr = zod.index(cmd[4])
    target = d[cmd[7]]
    t = target % 12
    if cmd[3] == "next":
        diff = yr - t
        if yr <= t:
            diff = yr + 12 - t
        d[new] = target + diff
    else:
        diff = t - yr
        if t <= yr:
            diff = t + 12 - yr
        d[new] = target - diff
    # print("\n", diff, yr, t)
# print(d)
print(abs(d["Bessie"] - d["Elsie"]))

File: USACO/2021/Feburary/Bronze/yoc.cpp
/*
ID: william234
TASK: yoc
LANG: C++
*/
#define PROGRAM_NAME "yoc"
#include <fstream>
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <algorithm>

#define DEBUG 0
#define USE_FILE 0
#define PARALLEL 1
#if PARALLEL
#pragma GCC optimize("Ofast")
#pragma GCC target("avx,avx2,fma")
#endif
#if USE_FILE
#define rin fin
#define rout fout
#define MAIN_FILE_HEADER                \
    ofstream fout(PROGRAM_NAME ".out"); \
    ifstream fin(PROGRAM_NAME ".in");
#else
#define rin cin
#define rout cout
#define MAIN_FILE_HEADER         \
    ios::sync_with_stdio(false); \
    cin.tie(nullptr);
#endif
#if DEBUG
#define IFDEBUG(stmt) stmt;
#define DBGOUT(stmt) cout << stmt;
#else
#define IFDEBUG(stmt)
#define DBGOUT(stmt)
#endif
#define ll long long

using namespace std;
string zod[] {"Ox", "Tiger", "Rabbit", "Dragon", "Snake", "Horse", "Goat", "Monkey", "Rooster", "Dog", "Pig", "Rat"};

int main() {
    MAIN_FILE_HEADER
    
    return 0;
}